### PR TITLE
MudJsonTreeView: Uses a MudTreeView to View JSON Text

### PR DESCRIPTION
**Pull request by jperson2000**
This update adds a new `MudJsonTreeView` component which can display JSON text using a `MudTreeView` component.  Supported features include:

* JSON string, GUID, date, number, and boolean values
* JSON arrays

This is a component which I use in production at my regular job, so it's been debugged and tested.  Let me know if anything else is needed.  Thanks!

### CODE REVIEW

File: CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.AspNetCore.Components;
+using System.Text.Json.Nodes;
+
+namespace MudExtensions;
+
+/// <summary>
+/// Represents a tree view which displays a snippet of JSON.
+/// </summary>
+public partial class MudJsonTreeView : ComponentBase
```
Comment (mckaragoz) [inline CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor.cs]: Shouldn't it be `MudComponentBase`

File: CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor [hunk 2]
```diff
@@ -0,0 +1,5 @@
+﻿@namespace MudExtensions
+
+<MudTreeView T="string" Dense="@Dense" Hover="@Hover">
```
Comment (mckaragoz) [inline CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor]: Do we need other parameters like `Class` and `Style`?

**Comment by mckaragoz** [comment]
Thanks for all effort, is it only for display or users can edit value also?

And we should add this new component also in docs. Need an example 🙂

File: CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor (hunk 2 above)
Comment (jperson2000) [reply CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor]: Yes, that sounds good.  Let me know if you think of any others you'd like to see.

**Comment by jperson2000** [comment]
> Thanks for all effort, is it only for display or users can edit value also?
> 
> And we should add this new component also in docs. Need an example 🙂

You're welcome, and I appreciate your PR reviews!  Currently it's only for display, and I'll get an example written up here soon.

File: CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor.cs (hunk 1 above)
Comment (jperson2000) [reply CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor.cs]: Ok, this is now `MudComponentBase`

File: CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor (hunk 2 above)
Comment (jperson2000) [reply CodeBeam.MudBlazor.Extensions/Components/JsonTreeView/MudJsonTreeView.razor]: Good catch; I've added `Class`, `Style`, `UserAttributes`, and an `OnJsonChanged` event.

**Comment by jperson2000** [comment]
> Thanks for all effort, is it only for display or users can edit value also?
> 
> And we should add this new component also in docs. Need an example 🙂

Hi @mckaragoz !  I added an example, but for some reason I get a 404 trying to open the example so I can debug it.   I bet I'm missing something simple.   Can you help me get the example page to display?

**Comment by mckaragoz** [comment]
> Hi @mckaragoz !  I added an example, but for some reason I get a 404 trying to open the example so I can debug it.   I bet I'm missing something simple.   Can you help me get the example page to display?

It should be Docs.csproj file, there are new lines about ItemContent, when we remove them it should be visible

**Comment by jperson2000** [comment]
> > Hi @mckaragoz !  I added an example, but for some reason I get a 404 trying to open the example so I can debug it.   I bet I'm missing something simple.   Can you help me get the example page to display?
> 
> It should be Docs.csproj file, there are new lines about ItemContent, when we remove them it should be visible

Thanks @mckaragoz !  That was it.  The example now functions.

**Comment by Sudheesh-TalentzQ** [comment]
Is this feature going to be added

**Comment by mckaragoz** [comment]
Thanks 🥇 

@Sudheesh-TalentzQ yes it's coming on MudExtensions v7

**Pull request merged**